src/shader.cpp

```diff
@@ -78,7 +78,11 @@ bool Shader::compileSourceCode(const QByteArray& src)
     {
         if (!re.exactMatch(lines[lineIdx]))
             continue;
-        QByteArray typeStr = re.cap(1).toAscii();
+#ifdef DISPLAZ_USE_QT4
+        QByteArray typeStr = re.cap(1).toAscii().constData();
+#else
+        QByteArray typeStr = re.cap(1).toLatin1().constData();
```

Comment:
Qt4 has `toLatin1()` - maybe we just use that, or possibly better `toUtf8()`?  There shouldn't be any strange characters here, it's glsl source after all.

For everywhere we have the choice and must convert QString to bytes, I'd like to use utf8 for C strings in preference to the other possible encodings.  Again, not really relevant here I expect since this is glsl source.